### ISSUE TITLE
Button has double keyboard focused borders inside DetailsList

### ISSUE THREAD

**Issue by adriantran** [OfficeDev/office-ui-fabric-react#3785]
### Bug Report

- __Package version(s)__: Latest
- __Browser and OS versions__: All

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No)
No (maybe later)

Requested priority: (Blocking, High, Normal, Low) 
Normal

Products/sites affected: (if applicable)
TFS

#### Describe the issue:

Buttons have double keyboard focus borders when composed inside of DetailsList

#### Actual behavior:

Double keyboard focus borders (outline from Button, having white for border, and reverted black borders from DetailsList cell.

e.g.
.ms-Fabric.is-focusVisible .cell_43fb9981 [data-is-focusable=true]:focus:after {
    content: '';
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    pointer-events: none;
    **border: 1px solid #666666;**
}
.ms-Fabric.is-focusVisible .root-38:focus::after {
    content: "";
    position: absolute;
    left: 0px;
    top: 0px;
    bottom: 0px;
    right: 0px;
    z-index: 1;
    border-width: 1px;
    border-style: solid;
    **border-color: rgb(255, 255, 255);**
    border-image: initial;
    **outline: rgb(102, 102, 102) solid 1px;**
}

#### Expected behavior:

Single keyboard focus border

### If applicable, please provide a codepen repro:
Try keyboard focus any button here to notice the double borders:

https://codepen.io/dzearing/pen/XRvXoL

**Comment by micahgodbolt** [comment]
issue stems from sass mixin using border, and buttons JS mixin using outline.

https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/office-ui-fabric-react/src/common/_focusBorder.scss#L54

Changing the sass would solve this issue, but we'd need to make sure it didn't have any other unintended consequences.

**Comment by adriantran** [comment]
(repeating from another thread)

Note that using outline has an issue--if the element with outline is inside a container that has overflow:hidden (likely for DetailsList, e.g. for css ellipsis), part of the outlines are missing.

**Comment by micahgodbolt** [comment]
Do you have a demo of this issue? The outline is actually on a pseudo element above the button, and not necessarily outside of the button's box boundaries.

**Comment by jordandrako** [comment]
@micahgodbolt Could this bug be a result of only partially converting to merged-styles? It seems the element is getting styled by both the Button focus (merged styles) and the DetailsList (sass). I think we should prioritize finishing the conversion.

**Comment by micahgodbolt** [comment]
well converting would fix it, but only because the sass mixing is working differently than the js mixin. the easier solution would be to make the two output the same css in the end.

**Comment by aditima** [comment]
As noted earlier, the fix is to convert the DetailsList component to use JS based styling. This is on our radar, but not funded yet.

**Comment by dzearing** [comment]
@aditima @ThomasMichon just fyi here; this one I believe is still an issue. I vaguely remember some work in trying to make cells focusable that may be causing this.

**Comment by micahgodbolt** [comment]
I know Rishu Mehndiratta asked me about this bug just today

**Comment by kenotron** [comment]
Hey all, I'm converting this to JS Styling and will account for it in my change.

**Comment by KevinTCoughlin** [comment]
@adriantran is this still an issue? I no longer see double borders in the original repro Codepen you shared pinned to latest OUFR.

**Comment by KevinTCoughlin** [comment]
When focused within a DetailsList row (the above Codepen) the Buttons look identical to how they behave in isolation so this issue seems to be fixed thanks to conversion to `mergeStyles` by @kenotron. Closing -- feel free to re-open if you believe it to still be an issue.
